### PR TITLE
Fix TypeError when using UPX compression

### PR DESCRIPTION
**Pull request by sebthom**
This fixes `TypeError: process_collected_binary() got an unexpected keyword argument 'strip'` as of **PyInstaller 6.0.0** when using UPX compression, because `strip`and `upx` parameters of `process_collected_binary` were renamed to `use_strip` and `use_upx`.

```py
+ python -O -m PyInstaller pyinstaller.spec --clean
1281 INFO: PyInstaller: 6.0.0
1281 INFO: Python: 3.12.0
1313 INFO: Platform: Windows-2022Server-10.0.20348-SP0
1344 INFO: UPX is available and will be used if enabled on build targets.
...
27594 INFO: checking PKG
27594 INFO: Building PKG because PKG-00.toc is non existent
27594 INFO: Building PKG (CArchive) foo.pkg
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\__main__.py", line 209, in <module>
    run()
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\__main__.py", line 189, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\__main__.py", line 61, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\building\build_main.py", line 1033, in main
    build(specfile, distpath, workpath, clean_build)
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\building\build_main.py", line 973, in build
    exec(code, spec_namespace)
  File "pyinstaller.spec", line 66, in <module>
    exe = EXE(pyz,
          ^^^^^^^^
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\building\api.py", line 606, in __init__
    self.pkg = PKG(
               ^^^^
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\building\api.py", line 238, in __init__
    self.__postinit__()
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\building\datastruct.py", line 184, in __postinit__
    self.assemble()
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\building\api.py", line 289, in assemble
    src_name = process_collected_binary(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\a\foo\foo\.venv\Lib\site-packages\PyInstaller\building\utils.py", line 208, in process_collected_binary
    src_name = process_collected_binary(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: process_collected_binary() got an unexpected keyword argument 'strip'
Error: Process completed with exit code 1.
```

### CODE REVIEW

**Review by rokm — APPROVED** [review]

**Comment by sebthom** [comment]
@rokm who else needs to approve for the build jobs to run?

**Comment by rokm** [comment]
> @rokm who else needs to approve for the build jobs to run?

Nobody, because we won't need to run the jobs for this one. The PR itself is waiting for approval from @bwoodsend, though.

**Comment by bwoodsend** [comment]
Yeah sorry, I saw this then forgot about it. Can we get a [news entry](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt). Looks good beyond that.

**Comment by rokm** [comment]
I've taken the liberty of adding the news fragment.

**Comment by sebthom** [comment]
@bwoodsend thanks a lot for merging the fix! I would appreciate a timely release as this is a blocker for our project. Is there any ETA for a 6.0.1 release?

**Comment by bwoodsend** [comment]
I think I'll just make a 6.1.0 release early. Patch releases require branching, pruning out the feature changes then reassembling the changelog back into one straight line – not something I do when I can avoid it. People are also waiting for #7968 albeit not for particularity good reasons...

**Pull request merged**
